numpy/core/src/multiarray/mapping.c

```diff
@@ -722,6 +723,38 @@ prepare_index(PyArrayObject *self, PyObject *index,
 
 
 /**
+ * Check if self has memory overlap with one of the index arrays, or with extra_op.
+ *
+ * @returns 1 if memory overlap found, 0 if not.
+ */
+NPY_NO_EXPORT int
+index_has_memory_overlap(PyArrayObject *self,
```

Comment:
It's used to checking the overlap between the array operated on, and the index array used in `ufunc.at`, for example
```
x = np.arange(100)
np.invert.at(x[::-1], x)  # <- oob memory access without x.copy() of the second argument
```